program-test/src/lib.rs

```diff
@@ -539,7 +583,7 @@ impl ProgramTest {
     ///
     /// Returns a `BanksClient` interface into the test environment as well as a payer `Keypair`
     /// with SOL for sending transactions
-    pub async fn start(self) -> (BanksClient, Keypair, Hash) {
+    pub async fn start(self) -> ProgramTestState {
```

Comment:
I'm definitely concerned about changing this signature and blowing up existing users.  I think we need to keep it for existing users and future users that just want to keep it simple.

Some ideas for a new method name for this signature:
1. `enter()` -- you enter the ProgramTextContext
2. `run()` -- confusing, do I "start" or "run"
3. `start_with_context()` -- maybe my favourite of the three, consistent with `start()` and the Rust norms of adding "with" methods that enhance the functionality of the base method